src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -121,33 +122,96 @@ public static IEnumerable<LowLevelClientMethod> BuildMethods(RestClientBuilder b
             }
         }
 
+        private Parameter CreateOptionsParameter(bool isRequired)
+        {
+            var clientOptionsType = ClientOptions.Type.WithNullable(true);
+            return new Parameter("options", "The options for configuring the client.", clientOptionsType, isRequired ? null : Constant.NewInstanceOf(clientOptionsType), false);
+        }
+
         private IEnumerable<ConstructorSignature> BuildPublicConstructors()
         {
             if (!_hasPublicConstructors)
             {
                 yield break;
             }
 
-            var clientOptionsType = ClientOptions.Type.WithNullable(true);
-            var clientOptionsParameter = new Parameter("options", "The options for configuring the client.", clientOptionsType, Constant.NewInstanceOf(clientOptionsType), false);
-
             if (Fields.CredentialFields.Count == 0)
             {
-                yield return BuildPublicConstructor(null, clientOptionsParameter);
+                foreach (var constructor in BuildPublicConstructor(null))
+                {
+                    yield return constructor;
+                }
             }
             else
             {
                 foreach (var credentialField in Fields.CredentialFields)
                 {
-                    yield return BuildPublicConstructor(credentialField, clientOptionsParameter);
+                    foreach (var constructor in BuildPublicConstructor(credentialField))
+                    {
+                        yield return constructor;
+                    }
                 }
             }
         }
 
-        private ConstructorSignature BuildPublicConstructor(FieldDeclaration? credentialField, Parameter clientOptionsParameter)
+        private ConstructorInitializer BuildConstructorInitializer(Parameter[] constructorParameters)
+        {
+            var initializer = constructorParameters.Select(parameter =>
+            {
+                if (parameter.DefaultValue == null)
+                {
+                    return FormattableStringFactory.Create("{0:I}", parameter.Name);
+                }
+
+                var defaultValue = parameter.DefaultValue.Value;
+                if (defaultValue.IsNewInstanceSentinel)
```

Comment:
Logic for constant writing is [more complicated](https://github.com/Azure/autorest.csharp/blob/eab7a29d375f7928066cacf19c8eb2e9f21f1dcf/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs#L387). We probably need to add helper method to [FormattableStringHelpers](https://github.com/Azure/autorest.csharp/blob/eab7a29d375f7928066cacf19c8eb2e9f21f1dcf/src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs), something like this:
```cs
public static FormattableString GetFormattable(this Constant constant)
{
    if (constant.Value == null)
    {
        // Cast helps the overload resolution
        return $"({constant.Type}){null:L}";
    }

    if (constant.IsNewInstanceSentinel)
    {
        return $"new {constant.Type}()";
    }

    if (!constant.Type.IsFrameworkType && constant.Value is EnumTypeValue enumTypeValue)
    {
        return $"{constant.Type}.{enumTypeValue.Declaration.Name}";
    }

    if (!constant.Type.IsFrameworkType && constant.Value is string enumValue)
    {
        return $"new {constant.Type}({enumValue:L})";
    }

    if (constant.Type.EqualsIgnoreNullable(typeof(DateTimeOffset)))
    {
        var d = (DateTimeOffset)constant.Value;
        d = d.ToUniversalTime();
        return $"new {typeof(DateTimeOffset)}({d.Year:L}, {d.Month:L}, {d.Day:L} ,{d.Hour:L}, {d.Minute:L}, {d.Second:L}, {d.Millisecond:L}, {typeof(TimeSpan)}.{nameof(TimeSpan.Zero)})";
    } 
    
    if (constant.Type.EqualsIgnoreNullable(typeof(byte[])))
    {
        var bytes = (byte[])constant.Value;
        return $"new byte[] {{{string.Join(", ", bytes)}}}";
    }
    
    if (constant.Type.Equals(typeof(ResourceType)))
    {
        return $"{((ResourceType)constant.Value).ToString():L}";
    }

    return $"{constant.Value:L}";
}
```

Then `WriteConstant` can be reduced to 
```cs
WriteConstant(this CodeWriter writer, Constant constant) => writer.Append(constant.GetFormattable());
```

Reply:
Yes, logic for constant is complicated. But here we only have constructor parameter constant. I didn't see conditions other than https://github.com/pshao25/autorest.csharp/blob/7c4faea10402aa45490cd461e26233294e1775ee/src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs#L61
Do you have examples?